tests/monad.cpp

```diff
@@ -228,3 +228,15 @@ TEST_CASE("operator|") {
         CHECK(val == Catch::Approx(4.61538));
     }
 }
+
+TEST_CASE("rsl::maybe_error") {
+    SECTION("Error") {
+        auto const exp = tl::expected<int, double>(tl::unexpected(0.1));
+        CHECK(rsl::maybe_error(exp));
+    }
+
+    SECTION("Value") {
+        auto const exp = tl::expected<int, double>(1);
+        CHECK_FALSE(rsl::maybe_error(exp));
+    }
```

Comment:
We need to test with zero parameters and more than 1 parameter since this is a parameter pack. Also test with parameter packs that have different numbers of unexpecteds.